File: AmiClean-Backend/AmiClean.Domain/Entities/Korisnik.cs
namespace AmiClean.Domain.Entities;

public class Korisnik
{
    public int ID_Korisnika { get; set; }
    public string Ime { get; set; } = null!;
    public string Prezime { get; set; } = null!;
    public string? Email { get; set; }
    public string Lozinka_Hash { get; set; } = null!;
    public string? Broj_Telefona { get; set; }
    public string? Adresa_Stanovanja { get; set; }
    public bool Aktivan { get; set; } = true;

    public virtual ICollection<Narudzba> Narudzbe { get; set; } = new List<Narudzba>();
    public virtual ICollection<Notifikacija> Notifikacije { get; set; } = new List<Notifikacija>();
    public virtual ICollection<Recenzija> Recenzije { get; set; } = new List<Recenzija>();
}

File: AmiClean-App/lib/features/auth/models/korisnik.dart
class Korisnik {
  const Korisnik({
    required this.id,
    required this.ime,
    required this.prezime,
    this.email,
    this.brojTelefona,
    this.adresaStanovanja,
    this.aktivan = true,
  });

  final int id;
  final String ime;
  final String prezime;
  final String? email;
  final String? brojTelefona;
  final String? adresaStanovanja;
  final bool aktivan;

  factory Korisnik.fromJson(Map<String, dynamic> json) {
    return Korisnik(
      id: json['id'] as int? ??
          json['id_Korisnika'] as int? ??
          json['iD_Korisnika'] as int? ??
          0,
      ime: json['ime'] as String,
      prezime: json['prezime'] as String,
      email: json['email'] as String?,
      brojTelefona:
          json['brojTelefona'] as String? ?? json['broj_Telefona'] as String?,
      adresaStanovanja: json['adresaStanovanja'] as String? ??
          json['adresa_Stanovanja'] as String?,
      aktivan: json['aktivan'] as bool? ?? true,
    );
  }

  String get punoIme => '$ime $prezime';
}
